src/order/max.lean

```diff
@@ -184,3 +184,42 @@ protected lemma is_max.eq_of_le (ha : is_max a) (h : a ≤ b) : a = b := h.antis
 protected lemma is_max.eq_of_ge (ha : is_max a) (h : a ≤ b) : b = a := h.antisymm' $ ha h
 
 end partial_order
+
+section prod
+variables [preorder α] [preorder β] {a a₁ a₂ : α} {b b₁ b₂ : β} {x y : α × β}
+
+lemma is_bot.prod_mk (ha : is_bot a) (hb : is_bot b) : is_bot (a, b) := λ c, ⟨ha _, hb _⟩
+lemma is_top.prod_mk (ha : is_top a) (hb : is_top b) : is_top (a, b) := λ c, ⟨ha _, hb _⟩
+lemma is_min.prod_mk (ha : is_min a) (hb : is_min b) : is_min (a, b) := λ c hc, ⟨ha hc.1, hb hc.2⟩
+lemma is_max.prod_mk (ha : is_max a) (hb : is_max b) : is_max (a, b) := λ c hc, ⟨ha hc.1, hb hc.2⟩
```

Comment:
Out of curiosity, can these be proved with weaker assumptions? eg does `is_max a`, `is_top b` imply `is_top (a, b)` (of course without the directness assumptions)

Reply:
I was wondering the same, but it really seems like it cannot. Consider `α` to be `bool` with the discrete order and `β := ℕ`. Then `is_min ff`, `is_bot 0`, but `¬ is_top (ff, 0)` because `¬ (ff, 0) ≤ (tt, 0)`.